src/backend/core/signals.py

```
@@ -42,11 +49,88 @@ def suppress_account_webhooks():
+@receiver(pre_save, sender=ServiceSubscription)
+def capture_proconnect_change(sender, instance, **kwargs):
+    """Record whether this save changes the subscription's routed domains.
+
+    Lets :func:`_sync_proconnect` skip the api-partenaires push when a save touches
+    neither ``is_active`` nor the domain list (e.g. an unrelated metadata edit).
+    """
+    if instance.service.type != "proconnect":
+        return
+    # Stash a transient flag on the instance for the post_save handler to read.
+    # pylint: disable=protected-access
+    if instance._state.adding or instance.pk is None:
```

Ruff (0.16.3)

[error] Private member accessed: `_state`

(SLF001)